libvast/src/system/index.cpp

```diff
@@ -1237,32 +1284,81 @@ index(index_actor::stateful_pointer<index_state> self,
-                  const auto* partition_legacy
-                    = partition->partition_as_legacy();
-                  for (const auto* partition_stats :
-                       *partition_legacy->type_ids()) {
-                    const auto* name = partition_stats->name();
-                    vast::ids ids;
-                    if (auto error
-                        = fbs::deserialize_bytes(partition_stats->ids(), ids)) {
-                      rp.deliver(caf::make_error(ec::format_error,
-                                                 "could not deserialize "
-                                                 "ids: "
-                                                   + render(error)));
+                  if (chunk->size() >= FLATBUFFERS_MAX_BUFFER_SIZE) {
+                    VAST_WARN("failed to load partition for deletion at {} "
+                              "because its size of {} exceeds the maximum "
+                              "allowed size of {}. The index statistics will "
+                              "be incorrect until the database has been "
+                              "rebuilt and restarted",
+                              path, chunk->size(), FLATBUFFERS_MAX_BUFFER_SIZE);
+                    // !!! Fallback path. We try to erase the corresponding
+                    // segment store file if it exists.
+                    // TODO: Also attempt to erase parquet stores once that
+                    // implementation is merged.
+                    auto store_path = store_path_for_partition(partition_id);
+                    self
+                      ->request<caf::message_priority::high>(
+                        self->state.filesystem, caf::infinite, atom::erase_v,
+                        path)
+                      .then(
+                        [self, partition_id](atom::done) {
+                          VAST_DEBUG("{} erased partition {} from filesystem",
+                                     *self, partition_id);
+                        },
+                        [self, partition_id, path](const caf::error& err) {
+                          VAST_WARN("{} failed to erase partition {} at {}: {}",
+                                    *self, partition_id, path, err);
+                        });
+                    self
+                      ->request<caf::message_priority::high>(
+                        self->state.filesystem, caf::infinite, atom::erase_v,
+                        store_path)
+                      .then(
+                        [self, partition_id, store_path](atom::done) {
+                          VAST_DEBUG("{} erased partition store {} from "
+                                     "filesystem",
+                                     *self, partition_id);
+                        },
+                        [self, partition_id,
+                         store_path](const caf::error& err) {
+                          VAST_DEBUG("{} failed to erase store {} at {}, "
```

Comment:
We're in a very special case recovery handler, so these can probably be safely higher than `DEBUG()`